nomad/drainer/drain_heap.go

```diff
@@ -96,7 +91,7 @@ func (d *deadlineHeap) watch() {
 			continue
 		}
 
-		if !deadline.Equal(nextDeadline) {
+		if deadline.IsZero() || !deadline.Equal(nextDeadline) {
```

Comment:
Add a comment to explain why either of these cases is valid.

Reply:
Done.